backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/ElasticsearchIndexFieldTypeFactoryImpl.java

```diff
@@ -234,6 +235,16 @@ public ScaledNumberIndexFieldTypeOptionsStep<?, BigInteger> asBigInteger() {
 		return new ElasticsearchBigIntegerIndexFieldTypeOptionsStep( this, typeDefaultsProvider );
 	}
 
+	@Override
+	public VectorFieldTypeOptionsStep<?, byte[]> asByteVector() {
+		throw new UnsupportedOperationException( "Implement me" );
+	}
+
+	@Override
+	public VectorFieldTypeOptionsStep<?, float[]> asFloatVector() {
+		throw new UnsupportedOperationException( "Implement me" );
+	}
```

Comment:
Come on, please think about the people who'll encounter the exception...

```suggestion
	@Override
	public VectorFieldTypeOptionsStep<?, byte[]> asByteVector() {
		throw new UnsupportedOperationException( "The Elasticsearch backend does not support vector field yet" );
	}

	@Override
	public VectorFieldTypeOptionsStep<?, float[]> asFloatVector() {
		throw new UnsupportedOperationException( "The Elasticsearch backend does not support vector field yet" );
	}
```

Reply:
This wasn't intended for the users 🙊  I was planning to move all of this vector-specific stuff under Lucene extensions, but for now, it would just make more noise, so these are just to make the compiler happy.

Reply:
>  I was planning to move all of this vector-specific stuff under Lucene extensions

That would be nice, though I suspect you'll have a hard time doing that, as you need to call these methods from the POJO module, which cannot depend on the Lucene backend. And also, this effort would be a bit wasted because we'll most likely implement the same thing for Elasticsearch/OpenSearch in the near future.

IMO don't bother and just leave self-explanatory exceptions.